cluster/gce/util.sh

```diff
@@ -1265,8 +1265,12 @@ function test-setup {
   until gcloud compute firewall-rules describe --project "${PROJECT}" "${NODE_TAG}-${INSTANCE_PREFIX}-http-alt" 2> /dev/null || [ $(($start + 1200)) -lt `date +%s` ]
   do sleep 5
   done
-  # Check if the firewall rule exists and fail if it does not.
-  gcloud compute firewall-rules describe --project "${PROJECT}" "${NODE_TAG}-${INSTANCE_PREFIX}-http-alt"
+  # Note, determine if we timed out, or if we found the firewall, you might think we could just
+  # check with gcloud, but sometimes that call flakes.  https://github.com/kubernetes/kubernetes/issues/20306
+  if [ $(($start + 1200)) -lt `date +%s` ]; then
```

Comment:
actually, maybe just rework the check entirely: replace the `until` on L1265 with something like
```bash
while ! gcloud compute firewall-rules describe --project "${PROJECT}" "${NODE_TAG}-${INSTANCE_PREFIX}-http-alt" 2> /dev/null; do
  if [[ $(($start + 1200)) -lt `date +%s` ]]; then
    echo -e "${color_red}Failed to create firewall ${NODE_TAG}-${INSTANCE_PREFIX}-http-alt in ${PROJECT}" >&2
    exit 1
  fi
  sleep 5
done

and similarly replace the loop to check the nodeports firewall just after this.